src/manifolds/Euclidean.jl

```diff
@@ -132,6 +132,39 @@ embed(::Euclidean, p)
 
 embed!(::Euclidean, q, p) = copyto!(q, p)
 
+@doc raw"""
+    embed(::EmbeddedManifold{𝔽,Euclidean{𝔽},Euclidean{𝔽},ET}, p)
+
+A [`Euclidean`](@ref) `M` manifold can be embedded into a [`Euclidean`](@ref) `N`
+if the length of the array dimension and representation size is elementwise less or equal.
+The remainder is filles with zeros for sure.
+"""
+function embed(M::EmbeddedManifold{𝔽,Euclidean{n,𝔽},Euclidean{m,𝔽},ET}, p) where {n,m,𝔽,ET}
+    q = allocate(p, representation_size(M.embedding))
+    embed!(M, q, p)
+    return q
+end
+
+function embed!(
+    M::EmbeddedManifold{𝔽,Euclidean{nL,𝔽},Euclidean{mL,𝔽},ET},
```

Comment:
I think we don't have to enforce the two `Euclidean` spaces being over the same number system? Invalid conversions will fail during assignment anyway.

Reply:
Yes, we sure can do that. And embedding from R to C should then just work, that's neat.